src/routes/ConfigureQuiz/QuestionCard.tsx

```diff
@@ -93,43 +94,73 @@ export default function QuestionCard({
     updateQuestionData(idx, question);
     await updateQuizData();
     setActiveQuestionIndex(null);
-    setExpandedQuestionIndex(idx);
+    setExpandedQuestionId(question.questionId);
 
     showAlert({
       message: 'Question has been saved successfully.',
       type: 'success',
     });
   }
 
+  function onQuestionSwap(data) {
+    const updatedQuestionIds = data.map((x) => x.questionId).join(',');
+    const initialQuestionIds = questions.map((x) => x.questionId).join(',');
+
+    if (initialQuestionIds !== updatedQuestionIds) {
+      replace(
+        data.map((x) => {
+          delete x.id;
+          return x;
+        }),
```

Comment:
Did not understand this part. What is going on here?

Reply:
for ReactSortable, I need a `id` for each question, So in below code I'm adding `id` key in ReactSortable list data. and here I'm removing `id` key before updating questions data

Reply:
Adding comments